src/libmeasurement_kit/dns/query_impl.hpp

```diff
@@ -214,10 +214,10 @@ template <MK_MOCK(evdns_base_free), MK_MOCK(evdns_base_resolve_ipv4),
         MK_MOCK(evdns_base_resolve_ipv6), MK_MOCK(evdns_base_resolve_reverse),
         MK_MOCK(evdns_base_resolve_reverse_ipv6), MK_MOCK(inet_pton)>
 void query_impl(QueryClass dns_class, QueryType dns_type, std::string name,
-        Callback<Error, Message> cb, Settings settings,
+        Callback<Error, Var<Message>> cb, Settings settings,
         Var<Reactor> reactor) {
 
-    Message message;
+    Var<Message> message(new Message);
```

Comment:
This is passed to `QueryContext()` and so all other uses of `message` through `context` look good